### PR TITLE
Compatibility/bug/warning fixes

### PR DESCRIPTION
**Pull request by drvink**
See individual commits.

### CODE REVIEW

**Comment by drvink** [comment]
FYI, I also had to remove the references to `CallGraphBuilder.cpp`, `JsonParser.cpp`, and `JsonParser.h` from the VS project file in order to build the plugin, but I haven't included a commit for that here, since I don't know if the fact that they're missing from the repo is intentional or not (though I assume it is, since it compiled successfully without them).

**Comment by trou** [comment]
FYI it compiles and seems to work on Linux.

File: src/HexRaysCodeXplorer/Common.h [hunk 1]
```diff
@@ -41,6 +41,7 @@
 #include "Linux.h"
 #endif
 
+#define USE_DANGEROUS_FUNCTIONS
```
Comment (trou) [inline src/HexRaysCodeXplorer/Common.h]: Note that *nix targets use the Makefile to define this, it should be updated if you declare it unconditionally.

**Comment by trou** [comment]
FWIW I've pushed more fixes/cleanups here: https://github.com/trou/HexRaysCodeXplorer/tree/fixes_more

File: src/HexRaysCodeXplorer/Common.h (hunk 1 above)
Comment (trou) [inline src/HexRaysCodeXplorer/Common.h]: My branch `fixes_more` removes the `-DUSE_DANGEROUS_FUNCTIONS`

**Comment by drvink** [comment]
@trou I've added your commits to this PR

**Comment by trou** [comment]
Thanks, make sure it compiles on Windows though :)

**Comment by drvink** [comment]
Works on Windows!

**Comment by drvink** [comment]
Does this PR need anything more in order to be merged?

**Comment by matrosov** [comment]
Guys sorry for the delay. Thanks for the fixes!

**Pull request merged**